crates/compiler/load_internal/src/file.rs

```diff
@@ -1317,8 +1317,13 @@ fn handle_root_type<'a>(
 
         use HeaderType::*;
 
+        let use_main = match header_type {
+            Package { .. } | App { .. } | Platform { .. } => true,
+            Module { .. } | Builtin { .. } | Hosted { .. } => false,
+        };
```

Comment:
maybe localize to the usage site

Reply:
Couldn't find a way due to borrow issue without introducing clones